src/System.Text.JsonLab/System/Text/Json/JsonWriterUtf8.cs

```diff
@@ -565,10 +565,32 @@ public void WriteValue(Guid value)
         /// </summary>
         public void WriteNull()
         {
-            //TODO: Optimize, just like WriteValue(long value)
-            WriteItemSeperatorUtf8();
-            _firstItem = false;
-            WriteSpacingUtf8();
+            int bytesNeeded = (_firstItem ? 0 : 1) + (_prettyPrint ? 2 + (_indent + 1) * 2 : 0);
+            Span<byte> byteBuffer = EnsureBuffer(bytesNeeded);
+            int idx = 0;
+            if (_firstItem)
+            {
+                _firstItem = false;
+            }
+            else 
+            {
+                byteBuffer[idx++] = JsonConstants.ListSeperator;
+            }
+
+            if (_prettyPrint) 
+            {
+                int indent = _indent;
+                byteBuffer[idx++] = JsonConstants.CarriageReturn;
+                byteBuffer[idx++] = JsonConstants.LineFeed;
+
+                while (indent-- >= 0)
+                {
+                    byteBuffer[idx++] = JsonConstants.Space;
+                    byteBuffer[idx++] = JsonConstants.Space;
+                }
+            }
+
+            _bufferWriter.Advance(bytesNeeded);
 
             WriteJsonValue(JsonConstants.NullValue);
```

Comment:
I think we can remove the WriteJsonValue call as well (for JsonWriterUtf8).
The WriteJsonValue just does a TryCopyTo call and we already know the length of JsonConstants.NullValue:
https://github.com/dotnet/corefxlab/blob/3b9952e99cbd50c9018e7976b31e08820b821a57/src/System.Text.Primitives/System/Text/SymbolTable/SymbolTable_utf8.cs#L51-L63

So, to avoid unnecessary calls to Advance, let's change this to something like:
```C#
JsonConstants.NullValue.CopyTo(byteBuffer.Slice(idx));
```